__tests__/hooks/use-display-currency.spec.tsx

```diff
@@ -0,0 +1,175 @@
+                  symbol: "¥",
+                  fractionDigits: 0,
+                  __typename: "Currency",
+                },
+              ],
+            },
+          },
+        },
+      ]
+
+      const wrapper = ({ children }: PropsWithChildren) => (
+        <IsAuthedContextProvider value={true}>
+          <MockedProvider mocks={mocksJpy}>{children}</MockedProvider>
+        </IsAuthedContextProvider>
+      )
+      const { result, waitForNextUpdate } = renderHook(() => useDisplayCurrency(), {
+        wrapper,
+      })
+
+      await waitForNextUpdate()
+
+      const res = result.current.moneyAmountToMajorUnitOrSats({
+        amount: 100,
+        currency: "DisplayCurrency",
+      })
+
+      expect(res).toBe(100)
+    })
+
+    it("with 2 digits", async () => {
+      const wrapper = ({ children }: PropsWithChildren) => (
+        <IsAuthedContextProvider value={true}>
+          <MockedProvider mocks={mocks}>{children}</MockedProvider>
+        </IsAuthedContextProvider>
+      )
+      const { result, waitForNextUpdate } = renderHook(() => useDisplayCurrency(), {
+        wrapper,
+      })
+
+      await waitForNextUpdate()
+
+      const res = result.current.moneyAmountToMajorUnitOrSats({
+        amount: 10,
+        currency: "DisplayCurrency",
+      })
+
+      expect(res).toBe(0.1)
+    })
+  })
+
+  it("unAuthed should return default value", async () => {
+    const wrapper = ({ children }: PropsWithChildren) => (
+      <IsAuthedContextProvider value={false}>
+        <MockedProvider mocks={[]}>{children}</MockedProvider>
+      </IsAuthedContextProvider>
+    )
+    const { result } = renderHook(() => useDisplayCurrency(), {
+      wrapper,
+    })
+
+    expect(result.current).toMatchObject({
+      fractionDigits: 2,
+      fiatSymbol: "$",
+      displayCurrency: "USD",
+    })
+  })
+
+  it("authed but empty query should return default value", async () => {
+    const wrapper = ({ children }: PropsWithChildren) => (
+      <IsAuthedContextProvider value={true}>
+        <MockedProvider mocks={[]}>{children}</MockedProvider>
+      </IsAuthedContextProvider>
+    )
+    const { result, waitForNextUpdate } = renderHook(() => useDisplayCurrency(), {
+      wrapper,
+    })
+
+    expect(result.current).toMatchObject({
+      fractionDigits: 2,
+      fiatSymbol: "$",
+      displayCurrency: "USD",
+    })
+
+    await waitForNextUpdate()
+
+    expect(result.current).toMatchObject({
+      fractionDigits: 2,
+      fiatSymbol: "$",
+      displayCurrency: "USD",
+    })
+  })
+
+  it("authed should return NGN from mock", async () => {
+    const wrapper = ({ children }: PropsWithChildren) => (
+      <IsAuthedContextProvider value={true}>
+        <MockedProvider mocks={mocks}>{children}</MockedProvider>
+      </IsAuthedContextProvider>
+    )
+    const { result, waitForNextUpdate } = renderHook(() => useDisplayCurrency(), {
+      wrapper,
+    })
+
+    expect(result.current).toMatchObject({
```

Comment:
Do we want this to be our behavior of briefly using USD until the query is resolved or would it be better for it to just be undefined?

Reply:
yup I have been thinking through this. I think is the `{ loading }` value on the `useQuery` is true, we may want to return undefined.

it's subtle because:
- beyond the undefined, what should we show in that case UX wise?
- we have to separate the use case loading with the use case where the query has an error (value could still be undefined, but we should show something else on the screen)